tests/test_envs.py

```diff
@@ -149,6 +150,31 @@ def test_rollout_schema(self, env: gym.Env, env_name: str):
         else:
             envs.test_rollout_schema(env)
 
-    def test_render(self, env: gym.Env):
-        """Tests `render()` supports modes specified in environment metadata."""
-        envs.test_render(env, raises_fn=pytest.raises)
+    def test_render_modes(self, env_name: str):
+        """Tests that all render modes specifeid in the metadata work.
+
+        Note: we only check that no exception is thrown.
+        There is no test to see if something reasonable is rendered.
+        """
+        for mode in gym.make(env_name).metadata["render_modes"]:
+            # GIVEN
+            env = gym.make(env_name, render_mode=mode)
+            env.reset(seed=0)
+
+            # WHEN
+            if mode == "rgb_array" and not is_mujoco_env(env):
+                # The render should not change without calling `step()`.
+                # MuJoCo rendering fails this check, ignore -- not much we can do.
```

Comment:
Why is this even being tested?

Reply:
Tbh I don't know. It seems to have been introduced in 0431decf by @AdamGleave . Maybe he recalls why?

Reply:
I think we ran into some environments that didn't support all of the render modes declared.

In general it seems good to test as much of each environment as possible. If custom rendering is defined then we should test it.

Is there something about the Gymnasium port that makes this difficult to test? The test does not seem essential to include if it will be painful to do so, but I'd err towards keeping tests, whereas we should certainly delete implementation code that no longer has a clear use.

Reply:
My point was "this specific case seems to be excluded from the test, why does this test exist at all in the first place".

I'm just not sure why we would check whether renderers return the same frame when called twice, since this is not something that we are developing first-party? Unless there is custom rendering code somewhere that I haven't seen.